packages/runtime/src/templates/edge-shared/rsc-data.ts

```diff
@@ -44,7 +44,7 @@ export const getRscDataRouter = ({ routes: staticRoutes, dynamicRoutes }: Preren
   )
 
   const dynamicRouteMatcher = Object.values(dynamicRoutes)
-    .filter(({ dataRoute }) => dataRoute.endsWith('.rsc'))
+    .filter(({ dataRoute }) => dataRoute?.endsWith('.rsc'))
```

Comment:
As mentioned in the description of the PR this wasn't being caught during compilation of the Next runtime as this file is copied.